File: java/space-age/src/main/java/SpaceAge.java
class SpaceAge {

    private final double seconds;

    private static final double divider = 60 * 60 * 24 * 365.25;
    private static final double mercuryPeriod = 0.2408467;
    private static final double venusPeriod = 0.61519726;
    private static final double marsPeriod = 1.8808158;
    private static final double jupiterPeriod = 11.862615;
    private static final double saturnPeriod = 29.447498;
    private static final double uranusPeriod = 84.016846;
    private static final double neptunePeriod = 164.79132;

    SpaceAge(double seconds) {
        this.seconds = seconds;
    }

    double getSeconds() {
        return seconds;
    }

    double onEarth() {
        return this.seconds / divider;
    }

    double onMercury() {
        return onEarth() / mercuryPeriod;
    }

    double onVenus() {
        return onEarth() / venusPeriod;
    }

    double onMars() {
        return onEarth() / marsPeriod;
    }

    double onJupiter() {
        return onEarth() / jupiterPeriod;
    }

    double onSaturn() {
        return onEarth() / saturnPeriod;
    }

    double onUranus() {
        return onEarth() / uranusPeriod;
    }

    double onNeptune() {
        return onEarth() / neptunePeriod;
    }

}

File: kotlin/space-age/src/main/kotlin/SpaceAge.kt
class SpaceAge(private val seconds: Int) {

    companion object {
        const val YEAR_ON_EARTH_IN_SECS = 31_557_600

        const val MERCURY_ORBITAL_PERIOD = 0.2408467
        const val VENUS_ORBITAL_PERIOD = 0.61519726
        const val MARS_ORBITAL_PERIOD = 1.8808158
        const val JUPITER_ORBITAL_PERIOD = 11.862615
        const val SATURN_ORBITAL_PERIOD = 29.447498
        const val URANUS_ORBITAL_PERIOD = 84.016846
        const val NEPTUNE_ORBITAL_PERIOD = 164.79132
    }

    fun onEarth(): Double = seconds / YEAR_ON_EARTH_IN_SECS.toDouble()
    fun onMercury(): Double = onEarth() / MERCURY_ORBITAL_PERIOD
    fun onVenus(): Double = onEarth() / VENUS_ORBITAL_PERIOD
    fun onMars(): Double = onEarth() / MARS_ORBITAL_PERIOD
    fun onJupiter(): Double = onEarth() / JUPITER_ORBITAL_PERIOD
    fun onSaturn(): Double = onEarth() / SATURN_ORBITAL_PERIOD
    fun onUranus(): Double = onEarth() / URANUS_ORBITAL_PERIOD
    fun onNeptune(): Double = onEarth() / NEPTUNE_ORBITAL_PERIOD
}
